drivers/i2c/i2c_ite_enhance.c

```diff
@@ -0,0 +1,776 @@
+	/* Frequency setting */
+	uint8_t freq;
+	/* wait for stop bit interrupt */
+	uint8_t stop;
+};
+
+enum enhanced_i2c_transfer_direct {
+	TX_DIRECT,
+	RX_DIRECT,
+};
+
+enum enhanced_i2c_ctl {
+	/* Hardware reset */
+	E_HW_RST = 0x01,
+	/* Stop */
+	E_STOP = 0x02,
+	/* Start & Repeat start */
+	E_START = 0x04,
+	/* Acknowledge */
+	E_ACK = 0x08,
+	/* State reset */
+	E_STS_RST = 0x10,
+	/* Mode select */
+	E_MODE_SEL = 0x20,
+	/* I2C interrupt enable */
+	E_INT_EN = 0x40,
+	/* 0 : Standard mode , 1 : Receive mode */
+	E_RX_MODE = 0x80,
+	/* State reset and hardware reset */
+	E_STS_AND_HW_RST = (E_STS_RST | E_HW_RST),
+	/* Generate start condition and transmit slave address */
+	E_START_ID = (E_INT_EN | E_MODE_SEL | E_ACK | E_START | E_HW_RST),
+	/* Generate stop condition */
+	E_FINISH = (E_INT_EN | E_MODE_SEL | E_ACK | E_STOP | E_HW_RST),
+};
+
+enum enhanced_i2c_host_status {
+	/* ACK receive */
+	E_HOSTA_ACK = 0x01,
+	/* Interrupt pending */
+	E_HOSTA_INTP = 0x02,
+	/* Read/Write */
+	E_HOSTA_RW = 0x04,
+	/* Time out error */
+	E_HOSTA_TMOE = 0x08,
+	/* Arbitration lost */
+	E_HOSTA_ARB = 0x10,
+	/* Bus busy */
+	E_HOSTA_BB = 0x20,
+	/* Address match */
+	E_HOSTA_AM = 0x40,
+	/* Byte done status */
+	E_HOSTA_BDS = 0x80,
+	/* time out or lost arbitration */
+	E_HOSTA_ANY_ERROR = (E_HOSTA_TMOE | E_HOSTA_ARB),
+	/* Byte transfer done and ACK receive */
+	E_HOSTA_BDS_AND_ACK = (E_HOSTA_BDS | E_HOSTA_ACK),
+};
+
+enum i2c_reset_cause {
+	I2C_RC_NO_IDLE_FOR_START = 1,
+	I2C_RC_TIMEOUT,
+};
+
+static int i2c_parsing_return_value(const struct device *dev)
+{
+	struct i2c_enhance_data *data = dev->data;
+
+	if (!data->err)
+		return 0;
+
+	/* Connection timed out */
+	if (data->err == ETIMEDOUT)
+		return -ETIMEDOUT;
+
+	/* The device does not respond ACK */
+	if (data->err == E_HOSTA_ACK)
+		return -ENXIO;
+	else
+		return -EIO;
+}
+
+static int i2c_get_line_levels(const struct device *dev)
+{
+	const struct i2c_enhance_config *config = dev->config;
+	uint8_t *base = config->base;
+	int pin_sts = 0;
+
+	if (IT8XXX2_I2C_TOS(base) & IT8XXX2_I2C_SCL_IN) {
+		pin_sts |= I2C_LINE_SCL_HIGH;
+	}
+
+	if (IT8XXX2_I2C_TOS(base) & IT8XXX2_I2C_SDA_IN) {
+		pin_sts |= I2C_LINE_SDA_HIGH;
+	}
+
+	return pin_sts;
+}
+
+static int i2c_is_busy(const struct device *dev)
+{
+	const struct i2c_enhance_config *config = dev->config;
+	uint8_t *base = config->base;
+
+	return (IT8XXX2_I2C_STR(base) & E_HOSTA_BB);
+}
+
+static int i2c_bus_not_available(const struct device *dev)
+{
+	if (i2c_is_busy(dev) ||
+		(i2c_get_line_levels(dev) != I2C_LINE_IDLE)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static void i2c_reset(const struct device *dev)
+{
+	const struct i2c_enhance_config *config = dev->config;
+	uint8_t *base = config->base;
+
+	/* State reset and hardware reset */
+	IT8XXX2_I2C_CTR(base) = E_STS_AND_HW_RST;
+}
+
+/* Set clock frequency for i2c port D, E , or F */
+static void i2c_enhanced_port_set_frequency(const struct device *dev,
+					    int freq_khz)
```

Comment:
This parameter is now the frequency in Hz, not kHz.

Reply:
Done.